app/src/main/java/com/alexbar/layoutbasic/screen/confirm_payment/widgets/AddNewCreditCardDialog.kt

```diff
@@ -0,0 +1,121 @@
+package com.alexbar.layoutbasic.screen.confirm_payment.widgets
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.*
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.*
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.window.Dialog
+import com.alexbar.layoutbasic.ui.theme.Dimens.dimen_16_dp
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_cancel
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_full_name
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_month
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_number
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_save
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_year
+
+@Composable
+fun AddCreditCardDialog(
+    onDismissRequest: () -> Unit
+) {
+    var name by remember { mutableStateOf("") }
+    var number by remember { mutableStateOf("") }
+    var expirationMonth by remember { mutableStateOf("") }
+    var expirationYear by remember { mutableStateOf("") }
+
+    Dialog(onDismissRequest = onDismissRequest) {
+        Box(modifier = Modifier
+                .fillMaxSize()
+                .background(Color.Black.copy(alpha = 0.5f))
+        )  {
+            Box(modifier = Modifier
+                .clip(RectangleShape)
+                .fillMaxWidth()
+                .align(Alignment.Center)
+                .background(Color.White)
+            ) {
+                Column {
+                    CreditCardUI(name, number, expirationMonth, expirationYear)
+
+                    TextField(
+                        value = name,
+                        onValueChange = { name = it },
+                        label = { Text(add_credit_card_full_name) },
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(horizontal = dimen_16_dp),
+                        colors = TextFieldDefaults.colors(
+                            focusedContainerColor = Color.White,
+                            unfocusedContainerColor = Color.White
+                        ),
+                    )
+
+                    TextField(
+                        value = number,
+                        onValueChange = { number = it },
+                        label = { Text(add_credit_card_number) },
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(horizontal = dimen_16_dp),
+                        colors = TextFieldDefaults.colors(
+                            focusedContainerColor = Color.White,
+                            unfocusedContainerColor = Color.White
+                        ),
+                    )
```

Comment:
Something I didn't know how to implement was displaying the credit card number in a specific format as the user inputs it into the UI. The goal was to insert a space after every 4 digits to achieve a final format like this: xxxx xxxx xxxx xxxx. but I didn't know how to do it.